File: 2021/01/2.pl
#!/usr/bin/perl

@lines = <>;

# print @lines;

$pf=$lines[0];
$psum=$lines[0] + $lines[1] + $lines[2];


for $i ( 3 .. $#lines ) {

    print "$psum\n";
    $sum=$psum-$pf+$lines[$i];
    $pf=$lines[$i-2];
    if($sum>$psum){$incs++;}
    $psum=$sum;

}

print $incs;

File: 2022/03/2.py
#!/usr/bin/python3

p1=0

f=open('input')
lines=f.readlines()

for l in range(0,len(lines),3):

    cs=set(lines[l].strip()).intersection(set(lines[l+1]).intersection(set(lines[l+2])))

    c=''.join(cs)

    v=ord(c)
    if v>96:
        v-=96
    if v>64:
        v-=38

    p1+=v

print("p1:",p1)
